src/lib/layers/SpeedLimits.svelte

```diff
@@ -0,0 +1,94 @@
+<script lang="ts">
+  import { onMount, onDestroy } from "svelte";
+  import {
+    emptyGeojson,
+    overwriteSource,
+    overwriteLayer,
+    drawLine,
+  } from "../../maplibre_helpers";
+  import type { LineString } from "geojson";
+  import type {
+    DataDrivenPropertyValueSpecification,
+    GeoJSONSource,
+  } from "maplibre-gl";
+  import type { Feature } from "../../types";
+  import { lineWidth } from "../../colors";
+  import { map, gjScheme } from "../../stores";
+  import { type Remote } from "comlink";
+  import { type RouteInfo } from "../../worker";
+  import MapTooltips from "../common/MapTooltips.svelte";
+  import DiscreteLegend from "../common/DiscreteLegend.svelte";
+  import HelpIcon from "../common/HelpIcon.svelte";
+
+  export let routeInfo: Remote<RouteInfo>;
+  // Show along a route if specified, or show all otherwise
+  export let id: number | undefined;
+
+  let colors = ["#00AB4D", "#8ECA4D", "#F7BB00", "#BB0000"];
+  let speedLimitSteps = [30, 40, 50];
+
+  let source = "speed-limits";
+  let layer = "speed-limits";
+  const colorBySpeedLimit: DataDrivenPropertyValueSpecification<string> = [
+    "step",
+    ["get", "speed_limit"],
+    colors[0],
+    speedLimitSteps[0],
+    colors[1],
+    speedLimitSteps[1],
+    colors[2],
+    speedLimitSteps[2],
+    colors[3],
+  ];
+  // NOTE! There's only ever one source and layer with this name. This component
```

Comment:
This is the first time in ATIP we do something like this; all other sources and layers live for the duration of the app. So we need the `onDestroy` here.